### ISSUE TITLE
When entering Chinese information, the pop-up window does not follow the cursor

### ISSUE THREAD

**Issue by oriming** [extrawurst/gitui#1699]
**Describe the bug**
When entering Chinese information, the pop-up window does not follow the cursor.

**To Reproduce**
1. Commit
2. Enter submission information

**Expected behavior**

**Screenshots**

**Context (please complete the following information):**
 - `macOS 13.1`
 - GitUI `0.22.1`
 - Rust `1.69.0`
 - Alacritty `0.12.1`

https://github.com/extrawurst/gitui/assets/61300279/63c3b4f3-16ce-42d3-8f33-76b8c7358116

**Comment by extrawurst** [comment]
is that only the case for Chinese?

**Comment by oriming** [comment]
@extrawurst This is a program bug and has nothing to do with the language.

**Comment by eclairevoyant** [comment]
I can reproduce on Linux with other languages. The issue is the IME candidate input or candidate list should be aligned to the cursor, but in gitui it bounces between aligning to the bottom right corner of the modal and the top left of the app itself.

I assume this is not an upstream issue because other ratatui-based programs I tested (ttyper, systeroid) don't appear to have this behavior

**Comment by extrawurst** [comment]
is that still an issue with `0.25` and the new textarea implementation?

**Comment by ZCShou** [comment]
> is that still an issue with `0.25` and the new textarea implementation?

Yes. I tested in windows terminal on windows 11 and ubuntu 20.04 , see the following picture.

![image](https://github.com/extrawurst/gitui/assets/8591065/4ee6a860-8926-45fc-a24e-e9d1354a056f)
